src/operator/drop.ts

```diff
@@ -0,0 +1,20 @@
+import { Subscribable, Listener, Subscription } from '../interfaces'
+
+export class DropOperator<T> implements Subscribable<T> {
+  constructor (private amount: number, public _subscribable: Subscribable<T>) {}
+
+  subscribe (listener: Listener<T>): Subscription {
+    return this._subscribable.subscribe(new DropListener<T>(this.amount, listener))
+  }
```

Comment:
I would actually say that merging the 2 classes would have a negative performance impact. I opted to follow the most.js architecture here. The reason I would suspect negative perf is because at the point you merge the 2 classes you have to capture the `listener` from subscribe as `this.listener`

```typescript
export class DropOperator<T> implement Subscribable<T>, Listener<T> {
  private class listener: Listener<T> = null
  ...
  subscribe(listener: Listener<T>) {
    this.listener = listener // the type of this.listener has changed a will cause de-opt
  }
}
```